diff --git a/WisT.DemoWeb.API/DependenciesResolver/Bootstraper.cs b/WisT.DemoWeb.API/DependenciesResolver/Bootstraper.cs
index c5fd97f..923d78f 100644
--- a/WisT.DemoWeb.API/DependenciesResolver/Bootstraper.cs
+++ b/WisT.DemoWeb.API/DependenciesResolver/Bootstraper.cs
@@ -2,7 +2,9 @@
 using WisT.DemoWeb.API.Infrastructure;
 using WisT.DemoWeb.API.Services;
 using WisT.DemoWeb.Persistence.Control;
+using WisT.DemoWeb.Persistence.DataEntities;
 using WisT.Recognizer.Contracts;
+using System.Data.Entity.Infrastructure;
 
 namespace WisT.DemoWeb.API.DependenciesResolver
 {
@@ -14,6 +16,7 @@ namespace WisT.DemoWeb.API.DependenciesResolver
             services.AddTransient<IRegistrationService, RegistrationService>();
             services.AddTransient<ILoginService, LoginService>();
             services.AddTransient<IRootDirectoryProvider, RootDirectoryProvider>();
+            services.AddTransient<IDbContextFactory<WisTEntities>, WisTContextFactory>();
         }
     }
 }
diff --git a/WisT.DemoWeb.API/WisT.DemoWeb.API.csproj b/WisT.DemoWeb.API/WisT.DemoWeb.API.csproj
index e618f55..c51877a 100644
--- a/WisT.DemoWeb.API/WisT.DemoWeb.API.csproj
+++ b/WisT.DemoWeb.API/WisT.DemoWeb.API.csproj
@@ -5,14 +5,16 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.AspNetCore" Version="2.0.1" />
-    <PackageReference Include="Microsoft.AspNetCore.Mvc" Version="2.0.2" />
+    <PackageReference Include="EntityFramework" Version="6.2.0" />
+    <PackageReference Include="Microsoft.AspNetCore" Version="2.1.0" />
+    <PackageReference Include="Microsoft.AspNetCore.Mvc" Version="2.1.0" />
     <PackageReference Include="Microsoft.AspNetCore.StaticFiles" Version="2.0.1" />
   </ItemGroup>
 
   <ItemGroup>
     <ProjectReference Include="..\Recognizer\Wist.Recognizer.Contracts\WisT.Recognizer.Contracts.csproj" />
     <ProjectReference Include="..\Recognizer\Wist.Recognizer\WisT.Recognizer.Identifier.csproj" />
+    <ProjectReference Include="..\WisT.DemoWeb.Persistence.Entities\WisT.DemoWeb.Persistence.DataEntities.csproj" />
     <ProjectReference Include="..\WisT.DemoWeb.Persistence\WisT.DemoWeb.Persistence.Control.csproj" />
   </ItemGroup>
 
diff --git a/WisT.DemoWeb.Persistence.Entities/WisT.DemoWeb.Persistence.DataEntities.csproj b/WisT.DemoWeb.Persistence.Entities/WisT.DemoWeb.Persistence.DataEntities.csproj
index 39e3249..2bfea68 100644
--- a/WisT.DemoWeb.Persistence.Entities/WisT.DemoWeb.Persistence.DataEntities.csproj
+++ b/WisT.DemoWeb.Persistence.Entities/WisT.DemoWeb.Persistence.DataEntities.csproj
@@ -41,10 +41,53 @@
     <Reference Include="EntityFramework.SqlServer, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089, processorArchitecture=MSIL">
       <HintPath>..\packages\EntityFramework.6.1.3\lib\net45\EntityFramework.SqlServer.dll</HintPath>
     </Reference>
+    <Reference Include="Microsoft.Extensions.Configuration, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60, processorArchitecture=MSIL">
+      <HintPath>..\packages\Microsoft.Extensions.Configuration.2.1.0\lib\netstandard2.0\Microsoft.Extensions.Configuration.dll</HintPath>
+    </Reference>
+    <Reference Include="Microsoft.Extensions.Configuration.Abstractions, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60, processorArchitecture=MSIL">
+      <HintPath>..\packages\Microsoft.Extensions.Configuration.Abstractions.2.1.0\lib\netstandard2.0\Microsoft.Extensions.Configuration.Abstractions.dll</HintPath>
+    </Reference>
+    <Reference Include="Microsoft.Extensions.Configuration.EnvironmentVariables, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60, processorArchitecture=MSIL">
+      <HintPath>..\packages\Microsoft.Extensions.Configuration.EnvironmentVariables.2.1.0\lib\netstandard2.0\Microsoft.Extensions.Configuration.EnvironmentVariables.dll</HintPath>
+    </Reference>
+    <Reference Include="Microsoft.Extensions.Configuration.FileExtensions, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60, processorArchitecture=MSIL">
+      <HintPath>..\packages\Microsoft.Extensions.Configuration.FileExtensions.2.1.0\lib\netstandard2.0\Microsoft.Extensions.Configuration.FileExtensions.dll</HintPath>
+    </Reference>
+    <Reference Include="Microsoft.Extensions.Configuration.Json, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60, processorArchitecture=MSIL">
+      <HintPath>..\packages\Microsoft.Extensions.Configuration.Json.2.1.0\lib\netstandard2.0\Microsoft.Extensions.Configuration.Json.dll</HintPath>
+    </Reference>
+    <Reference Include="Microsoft.Extensions.FileProviders.Abstractions, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60, processorArchitecture=MSIL">
+      <HintPath>..\packages\Microsoft.Extensions.FileProviders.Abstractions.2.1.0\lib\netstandard2.0\Microsoft.Extensions.FileProviders.Abstractions.dll</HintPath>
+    </Reference>
+    <Reference Include="Microsoft.Extensions.FileProviders.Physical, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60, processorArchitecture=MSIL">
+      <HintPath>..\packages\Microsoft.Extensions.FileProviders.Physical.2.1.0\lib\netstandard2.0\Microsoft.Extensions.FileProviders.Physical.dll</HintPath>
+    </Reference>
+    <Reference Include="Microsoft.Extensions.FileSystemGlobbing, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60, processorArchitecture=MSIL">
+      <HintPath>..\packages\Microsoft.Extensions.FileSystemGlobbing.2.1.0\lib\netstandard2.0\Microsoft.Extensions.FileSystemGlobbing.dll</HintPath>
+    </Reference>
+    <Reference Include="Microsoft.Extensions.Primitives, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60, processorArchitecture=MSIL">
+      <HintPath>..\packages\Microsoft.Extensions.Primitives.2.1.0\lib\netstandard2.0\Microsoft.Extensions.Primitives.dll</HintPath>
+    </Reference>
+    <Reference Include="Newtonsoft.Json, Version=11.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL">
+      <HintPath>..\packages\Newtonsoft.Json.11.0.2\lib\net45\Newtonsoft.Json.dll</HintPath>
+    </Reference>
     <Reference Include="System" />
+    <Reference Include="System.Buffers, Version=4.0.2.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51, processorArchitecture=MSIL">
+      <HintPath>..\packages\System.Buffers.4.4.0\lib\netstandard2.0\System.Buffers.dll</HintPath>
+    </Reference>
     <Reference Include="System.ComponentModel.DataAnnotations" />
     <Reference Include="System.Data" />
     <Reference Include="System.Drawing" />
+    <Reference Include="System.Memory, Version=4.0.1.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51, processorArchitecture=MSIL">
+      <HintPath>..\packages\System.Memory.4.5.0\lib\netstandard2.0\System.Memory.dll</HintPath>
+    </Reference>
+    <Reference Include="System.Numerics" />
+    <Reference Include="System.Numerics.Vectors, Version=4.1.3.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
+      <HintPath>..\packages\System.Numerics.Vectors.4.4.0\lib\net46\System.Numerics.Vectors.dll</HintPath>
+    </Reference>
+    <Reference Include="System.Runtime.CompilerServices.Unsafe, Version=4.0.4.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
+      <HintPath>..\packages\System.Runtime.CompilerServices.Unsafe.4.5.0\lib\netstandard2.0\System.Runtime.CompilerServices.Unsafe.dll</HintPath>
+    </Reference>
     <Reference Include="System.Runtime.Serialization" />
     <Reference Include="System.Security" />
     <Reference Include="System.Xml" />
@@ -62,11 +105,14 @@
     <Compile Include="Migrations\Configuration.cs" />
     <Compile Include="User.cs" />
     <Compile Include="UserImage.cs" />
+    <Compile Include="WisTContextFactory.cs" />
     <Compile Include="WisTEntities.cs" />
   </ItemGroup>
   <ItemGroup>
     <None Include="App.config" />
-    <None Include="packages.config" />
+    <None Include="packages.config">
+      <SubType>Designer</SubType>
+    </None>
   </ItemGroup>
   <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
 </Project>
\ No newline at end of file
diff --git a/WisT.DemoWeb.Persistence.Entities/WisTEntities.cs b/WisT.DemoWeb.Persistence.Entities/WisTEntities.cs
index 77e9ed8..0103d28 100644
--- a/WisT.DemoWeb.Persistence.Entities/WisTEntities.cs
+++ b/WisT.DemoWeb.Persistence.Entities/WisTEntities.cs
@@ -5,8 +5,8 @@ namespace WisT.DemoWeb.Persistence.DataEntities
 
     public partial class WisTEntities : DbContext
     {
-        public WisTEntities()
-            : base("name=WisTEntities")
+        public WisTEntities(string connectionString)
+            : base(connectionString)
         {
             Database.SetInitializer(new MigrateDatabaseToLatestVersion<WisTEntities, Migrations.Configuration>());
         }
diff --git a/WisT.DemoWeb.Persistence.Entities/packages.config b/WisT.DemoWeb.Persistence.Entities/packages.config
index 8b3f03e..81815c8 100644
--- a/WisT.DemoWeb.Persistence.Entities/packages.config
+++ b/WisT.DemoWeb.Persistence.Entities/packages.config
@@ -1,4 +1,18 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
   <package id="EntityFramework" version="6.1.3" targetFramework="net462" />
+  <package id="Microsoft.Extensions.Configuration" version="2.1.0" targetFramework="net462" />
+  <package id="Microsoft.Extensions.Configuration.Abstractions" version="2.1.0" targetFramework="net462" />
+  <package id="Microsoft.Extensions.Configuration.EnvironmentVariables" version="2.1.0" targetFramework="net462" />
+  <package id="Microsoft.Extensions.Configuration.FileExtensions" version="2.1.0" targetFramework="net462" />
+  <package id="Microsoft.Extensions.Configuration.Json" version="2.1.0" targetFramework="net462" />
+  <package id="Microsoft.Extensions.FileProviders.Abstractions" version="2.1.0" targetFramework="net462" />
+  <package id="Microsoft.Extensions.FileProviders.Physical" version="2.1.0" targetFramework="net462" />
+  <package id="Microsoft.Extensions.FileSystemGlobbing" version="2.1.0" targetFramework="net462" />
+  <package id="Microsoft.Extensions.Primitives" version="2.1.0" targetFramework="net462" />
+  <package id="Newtonsoft.Json" version="11.0.2" targetFramework="net462" />
+  <package id="System.Buffers" version="4.4.0" targetFramework="net462" />
+  <package id="System.Memory" version="4.5.0" targetFramework="net462" />
+  <package id="System.Numerics.Vectors" version="4.4.0" targetFramework="net462" />
+  <package id="System.Runtime.CompilerServices.Unsafe" version="4.5.0" targetFramework="net462" />
 </packages>
\ No newline at end of file
diff --git a/WisT.DemoWeb.Persistence/LabelStorage.cs b/WisT.DemoWeb.Persistence/LabelStorage.cs
index ce23706..bc08be9 100644
--- a/WisT.DemoWeb.Persistence/LabelStorage.cs
+++ b/WisT.DemoWeb.Persistence/LabelStorage.cs
@@ -1,5 +1,6 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.Data.Entity.Infrastructure;
 using System.Linq;
 using WisT.DemoWeb.Persistence.Control.Excepsions;
 using WisT.DemoWeb.Persistence.DataEntities;
@@ -10,9 +11,16 @@ namespace WisT.DemoWeb.Persistence.Control
 {
     public class LabelStorage : ILabelStorage
     {
+        private IDbContextFactory<WisTEntities> _contextFactory;
+
+        public LabelStorage(IDbContextFactory<WisTEntities> contextFactory)
+        {
+            _contextFactory = contextFactory;
+        }
+
         public void Add(ILabel addObj)
         {
-            using (WisTEntities context = new WisTEntities())
+            using (WisTEntities context = _contextFactory.Create())
             {
                 User user = new User(addObj);
 
@@ -41,7 +49,7 @@ namespace WisT.DemoWeb.Persistence.Control
 
         public ILabel Get(IIdentifier id)
         {
-            using (WisTEntities context = new WisTEntities())
+            using (WisTEntities context = _contextFactory.Create())
             {
                 User user = new User();
 
@@ -62,7 +70,7 @@ namespace WisT.DemoWeb.Persistence.Control
 
         public IEnumerable<ILabel> GetAll()
         {
-            using (WisTEntities context = new WisTEntities())
+            using (WisTEntities context = _contextFactory.Create())
             {
                 return context.Users.ToList().Select(x => new DatabaseLabel(x)).ToList();
             }
diff --git a/WisT.DemoWeb/Startup.cs b/WisT.DemoWeb/Startup.cs
index 98ae628..3c03ee8 100644
--- a/WisT.DemoWeb/Startup.cs
+++ b/WisT.DemoWeb/Startup.cs
@@ -3,7 +3,9 @@ using Microsoft.AspNetCore.Hosting;
 using Microsoft.AspNetCore.SpaServices.Webpack;
 using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
 using WisT.DemoWeb.API.DependenciesResolver;
+using WisT.DemoWeb.API.Infrastructure;
 
 namespace WisT.DemoWeb
 {
@@ -19,13 +21,16 @@ namespace WisT.DemoWeb
         // This method gets called by the runtime. Use this method to add services to the container.
         public void ConfigureServices(IServiceCollection services)
         {
-            services.AddMvc();
+            services.AddMvc(config =>
+            {
+                config.Filters.Add(typeof(ExceptionFilter));
+            });
             services.AddSingleton<IConfiguration>(Configuration);
             Bootstraper.RegisterWisTDependencies(services);
         }
 
         // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
-        public void Configure(IApplicationBuilder app, IHostingEnvironment env)
+        public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerFactory logging)
         {
             if (env.IsDevelopment())
             {
@@ -41,6 +46,10 @@ namespace WisT.DemoWeb
                 app.UseExceptionHandler("/Home/Error");
             }
 
+            logging.AddConsole(Configuration.GetSection("Logging"));
+            logging.AddDebug();
+            logging.AddAzureWebAppDiagnostics();
+
             app.UseStaticFiles();
 
             app.UseMvc(routes =>
diff --git a/WisT.DemoWeb/WisT.DemoWeb.csproj b/WisT.DemoWeb/WisT.DemoWeb.csproj
index 65b1557..0eb1872 100644
--- a/WisT.DemoWeb/WisT.DemoWeb.csproj
+++ b/WisT.DemoWeb/WisT.DemoWeb.csproj
@@ -30,10 +30,12 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.AspNetCore" Version="2.0.1" />
-    <PackageReference Include="Microsoft.AspNetCore.Mvc" Version="2.0.2" />
-    <PackageReference Include="Microsoft.AspNetCore.SpaServices" Version="2.0.2" />
-    <PackageReference Include="Microsoft.AspNetCore.StaticFiles" Version="2.0.1" />
+    <PackageReference Include="Microsoft.AspNetCore" Version="2.1.0" />
+    <PackageReference Include="Microsoft.AspNetCore.Mvc" Version="2.1.0" />
+    <PackageReference Include="Microsoft.AspNetCore.SpaServices" Version="2.1.0" />
+    <PackageReference Include="Microsoft.AspNetCore.StaticFiles" Version="2.1.0-rc1-final" />
+    <PackageReference Include="Microsoft.Extensions.Logging.AzureAppServices" Version="2.1.0" />
+    <PackageReference Include="Microsoft.Extensions.Logging.TraceSource" Version="2.1.0" />
   </ItemGroup>
 
   <ItemGroup>
@@ -49,6 +51,15 @@
     </Reference>
   </ItemGroup>
 
+  <ItemGroup>
+    <Content Update="appsettings.Development.json">
+      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
+    </Content>
+    <Content Update="appsettings.json">
+      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
+    </Content>
+  </ItemGroup>
+
   <Target Name="DebugRunWebpack" BeforeTargets="Build" Condition=" '$(Configuration)' == 'Debug' And !Exists('wwwroot\dist') ">
     <!-- Ensure Node.js is installed -->
     <Exec Command="node --version" ContinueOnError="true">
diff --git a/WisT.DemoWeb/appsettings.json b/WisT.DemoWeb/appsettings.json
index 1b5b8f8..3a1358a 100644
--- a/WisT.DemoWeb/appsettings.json
+++ b/WisT.DemoWeb/appsettings.json
@@ -1,4 +1,7 @@
 {
+  "ConnectionStrings": {
+    "WisTEntities": "Server=(localdb)\\mssqllocaldb;Database=WisT;Trusted_Connection=True;MultipleActiveResultSets=true"
+  },
   "Logging": {
     "LogLevel": {
       "Default": "Warning"
